frontend/javascripts/viewer/view/components/input_component.tsx

```diff
@@ -18,120 +15,95 @@ type InputComponentState = {
  *   while it's focused (mainly necessary when mutating the value on arrow-keypresses)
  */
 
-class InputComponent extends React.PureComponent<InputProps, InputComponentState> {
-  inputRef = React.createRef<InputRef>();
-  static defaultProps: InputProps = {
-    onChange: _.noop,
-    placeholder: "",
-    value: "",
-    style: {},
-  };
-
-  state = {
-    currentValue: this.props.value,
-  };
-
-  getSnapshotBeforeUpdate(
-    _prevProps: InputProps,
-    _prevState: InputComponentState,
-  ): [number | null, number | null] {
-    // Remember the selection within the input before updating it.
-    try {
-      return [
-        // @ts-ignore
-        this.inputRef?.current?.input.selectionStart,
-        // @ts-ignore
-        this.inputRef?.current?.input.selectionEnd,
-      ];
-    } catch {
-      return [null, null];
+function InputComponent(props: InputProps) {
+  const {
+    title,
+    style,
```

Comment:
The previous default style is missing here
```suggestion
    style = {},
```

Reply:
but not 100% whether that necessary to keep

Reply:
I think it is not needed. Style will default to undefined. And thus FastTooltip will received undefined as style, which should also be find.